### PR TITLE
solution

### PR DESCRIPTION
**Pull request by abcdesign1**

### CODE REVIEW

**Review by arkushenko-anna — APPROVED** [review]
Perfect!

File: src/makeRobotAccountant.js [hunk 1]
```diff
@@ -6,7 +6,21 @@
...
+    store.count++;
+
+    if (store.count >= 4 && store.count % 2 === 0) {
+      return () => 'Bzzz... Error!';
+    }
+
+    return (y) => x + y;
+  };
+
+  return funcObj;
```
Comment (illiarerun) [inline src/makeRobotAccountant.js:23]: you can return this function right  away, no need for this variable
```suggestion
  return function(x) {
    store.count++;

    if (store.count >= 4 && store.count % 2 === 0) {
      return () => 'Bzzz... Error!';
    }

    return (y) => x + y;
  };
;
```

File: src/makeRobotAccountant.js [hunk 2]
```diff
@@ -6,7 +6,21 @@
...
+  store.count = 0;
+
+  const funcObj = function(x) {
+    store.count++;
+
+    if (store.count >= 4 && store.count % 2 === 0) {
+      return () => 'Bzzz... Error!';
+    }
+
+    return (y) => x + y;
```
Comment (illiarerun) [inline src/makeRobotAccountant.js:20]: you're returning function in both conditions, just create an inner function that will return you one of the values and return it

**Review by illiarerun — CHANGES_REQUESTED** [review]

File: src/makeRobotAccountant.js (hunk 1 above)
Comment (vadym4che) [reply src/makeRobotAccountant.js]: fixed

File: src/makeRobotAccountant.js (hunk 2 above)
Comment (vadym4che) [reply src/makeRobotAccountant.js]: refactored

hope I've got your advises right

image

File: src/makeRobotAccountant.js [hunk 3]
```diff
@@ -6,7 +6,21 @@
  */
 
 function makeRobotAccountant() {
-  // write code here
+  let count = 0;
+
+  const funcObj = function(x) {
```
Comment (sergii-nosachenko) [inline src/makeRobotAccountant.js:11]:
```suggestion
  return function(x) {
```

File: src/makeRobotAccountant.js [hunk 4]
```diff
@@ -6,7 +6,21 @@
...
+    count++;
+
+    return (y) => {
+      if (count >= 4 && count % 2 === 0) {
```
Comment (sergii-nosachenko) [inline src/makeRobotAccountant.js:15]:
```suggestion
      if (count > 3 && count % 2 === 0) {
```

**Review by sergii-nosachenko — CHANGES_REQUESTED** [review]

File: src/makeRobotAccountant.js (hunk 3 above)
Comment (vadym4che) [reply src/makeRobotAccountant.js:11]: fixed
got it
thank you for your advises

File: src/makeRobotAccountant.js (hunk 4 above)
Comment (vadym4che) [reply src/makeRobotAccountant.js]: got this point
fixed

**Review by sergii-nosachenko — APPROVED** [review]

**Pull request closed without merging**